src/transformers/modeling_funnel.py

```diff
@@ -0,0 +1,1529 @@
+
+        self.funnel = FunnelModel(config)
+        self.lm_head = nn.Linear(config.d_model, config.vocab_size)
+
+        self.init_weights()
+
+    def get_output_embeddings(self):
+        return self.lm_head
+
+    @add_start_docstrings_to_callable(FUNNEL_INPUTS_DOCSTRING.format("(batch_size, sequence_length)"))
+    @add_code_sample_docstrings(
+        tokenizer_class=_TOKENIZER_FOR_DOC,
+        checkpoint="funnel-small",
+        output_type=MaskedLMOutput,
+        config_class=_CONFIG_FOR_DOC,
+    )
+    def forward(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        token_type_ids=None,
+        inputs_embeds=None,
+        labels=None,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_dict=None,
+    ):
+        r"""
+        labels (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`, `optional`, defaults to :obj:`None`):
+            Labels for computing the masked language modeling loss.
+            Indices should be in ``[-100, 0, ..., config.vocab_size]`` (see ``input_ids`` docstring)
+            Tokens with indices set to ``-100`` are ignored (masked), the loss is only computed for the tokens with labels
+            in ``[0, ..., config.vocab_size]``
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.funnel(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        last_hidden_state = outputs[0]
+        prediction_logits = self.lm_head(last_hidden_state)
+
+        masked_lm_loss = None
+        if labels is not None:
+            loss_fct = CrossEntropyLoss()  # -100 index = padding token
+            masked_lm_loss = loss_fct(prediction_logits.view(-1, self.config.vocab_size), labels.view(-1))
+
+        if not return_dict:
+            output = (prediction_logits,) + outputs[1:]
+            return ((masked_lm_loss,) + output) if masked_lm_loss is not None else output
+
+        return MaskedLMOutput(
+            loss=masked_lm_loss,
+            logits=prediction_logits,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
+
+
+@add_start_docstrings(
+    """Funnel Transfprmer Model with a sequence classification/regression head on top (two linear layer on top of
+    the first timestep of the last hidden state) e.g. for GLUE tasks. """,
+    FUNNEL_START_DOCSTRING,
+)
+class FunnelForSequenceClassification(FunnelPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.funnel = FunnelBaseModel(config)
+        self.classifier = nn.Sequential(
```

Comment:
not a huge fan of `nn.Sequential()` here. Colud we maybe create a new `nn.Module` for this? 
I think this keeps the code more flexible and readable

Reply:
I don't see how creating a module makes it more readable, it's actually longer (you have to write the forward pass) while adding nothing of value. I can refactor this in a separate function if needed but I don't really get what the end user will gain in terms of flexibility. In both cases, they'll have to edit this file for modifications.

Reply:
My arguments are the following (don't have a very strong opinion on this though):

1) Creating a separate model is more in line with how it is done in other models (other models always create a seperate class for things like `classification_head`, `lm_head`, `mlm_head`. Because we never use `nn.Sequential` anywhere in the code, a user is "more" surprised by this new feature -> https://en.wikipedia.org/wiki/Principle_of_least_astonishment

2) IMO, it's less flexible in a sense that it's harder to tweak with special functions, like a residual connection between input first linear layer output or special mathematical functions for which a lambda layer would probably be needed

3) It's impossible to debug between `nn.Linear` and `nn.Tanh()` e.g. if one wants to quickly know the output of the layer, he would have to rewrite the whole `nn.Sequential` object

Reply:
Okey actually CTLR and Transfo-XL use `nn.Sequential` as well, but I don't think that these are our cleanest models

Reply:
All the arguments are sound, but apply to a complex model. This is just two linear layers with a tanh and a dropout in the middle. Anyhow, will refactor as a module since it's going to take less time than debating about it.